ci-operator/step-registry/hypershift/hostedcluster/destroy/hostedcluster/hypershift-hostedcluster-destroy-hostedcluster-commands.sh

```diff
@@ -40,19 +46,33 @@ fi
 echo Cluster successfully created at $createdAt
 
 echo "$(date) Deleting HyperShift cluster ${CLUSTER_NAME}"
-
-for _ in {1..10}; do
-  bin/hypershift destroy cluster aws \
-    --aws-creds=${AWS_GUEST_INFRA_CREDENTIALS_FILE}  \
+if [[ "${PLATFORM}" == "aws" ]]; then
+  for _ in {1..10}; do
+   bin/hypershift destroy cluster aws \
+     --aws-creds=${AWS_GUEST_INFRA_CREDENTIALS_FILE}  \
+     --name ${CLUSTER_NAME} \
+     --infra-id ${INFRA_ID} \
+     --region ${HYPERSHIFT_AWS_REGION} \
+     --base-domain ${DOMAIN} \
+     --cluster-grace-period 40m
+   if [ $? == 0 ]; then
+     break
+   else
+     echo 'Failed to delete the cluster, retrying...'
+   fi
+  done
+else
+  bin/hypershift destroy cluster powervs \
```

Comment:
Can we implement the same retry mechanism for powervs destroy as well?

Reply:
Sure, added!